### PR TITLE
PHOENIX-5546: TASK_TS being set as HConstants.LATEST_TIMESTAMP in SYSTEM.TASK table

### PR DESCRIPTION
**Pull request by ChinmaySKulkarni**
Please review @twdsilva @lhofhansl @kadirozde

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java [hunk 1]
```diff
@@ -776,8 +771,14 @@ private long validateAndGetServerTimestamp(TableRef tableRef, MultiRowMutationSt
         // If we're auto committing, we've already validated the schema when we got the ColumnResolver,
         // so no need to do it again here.
         PTable table = tableRef.getTable();
-        MetaDataMutationResult result = client.updateCache(table.getSchemaName().getString(), table.getTableName()
-                .getString());
+
+        // We generally don't re-resolve SYSTEM tables, but if it relies on ROW_TIMESTAMP, we must
```
Comment (twdsilva) [inline phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java]: Would it make sense to change the check we have here for system tables to also check if the system table doesn't have ROW_TIMESTAMP column instead?
https://github.com/apache/phoenix/blob/dd5551b4e8cc17e5bd587e6c4f5f256a8212345a/phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java#L606

Comment (ChinmaySKulkarni) [reply phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java]: @twdsilva I think that makes sense. I will update the PR. Thanks!

File: phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java [hunk 2]
```diff
@@ -725,6 +719,20 @@ public MetaDataMutationResult updateCache(PName origTenantId, String schemaName,
         return result;
     }
 
+    // Do not make rpc to getTable if
+    // 1. table is a system table that does not have a ROW_TIMESTAMP column OR
+    // 2. table was already resolved as of that timestamp OR
+    // 3. table does not have a ROW_TIMESTAMP column and age is less then UPDATE_CACHE_FREQUENCY
+    private boolean avoidRpcToGetTable(boolean alwaysHitServer, Long resolvedTimestamp,
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java]: @twdsilva made the change as per your review. Please take a look.

**Comment by ChinmaySKulkarni** [comment]
Thanks for the reviews @twdsilva @kadirozde

**Comment by gjacoby126** [comment]
JIRA resolved, closing

**Pull request closed without merging**